kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/develop/WatchMojoTest.java

```diff
@@ -84,6 +84,7 @@ void setUp(@TempDir Path temporaryFolder) throws Exception {
     when(mavenProject.getCompileClasspathElements()).thenReturn(Collections.emptyList());
     when(mavenProject.getDependencies()).thenReturn(Collections.emptyList());
     when(mavenProject.getDevelopers()).thenReturn(Collections.emptyList());
+    when(mavenProject.getBasedir()).thenReturn(temporaryFolder.toFile());
```

Comment:
is this related to refactor ?

Reply:
`SimpleDockerfileGenerator.isApplicable` method requires project base directory to be provided. `mavenProject` is mocked with deep stubs; when we invoke `mavenProject.getBasedir()` it is returning some mocked object which is causing NPE when this base directory is checked in  `SimpleDockerfileGenerator.isApplicable`